packages/publisher/index.ts

```diff
@@ -0,0 +1,226 @@
+            guildedCommands.forEach((entry) => {
+                const guildIds: string[] = entry.guildIds ?? []; 
+                if (guildIds) {
+                    guildIds.forEach((guildId) => {
+                        if (guildIdMap.has(guildId)) {
+                            guildIdMap.get(guildId)?.push(entry);
+                        } else {
+                            guildIdMap.set(guildId, [entry]);
+                        }
+                    });
+                }
+            });
+            for (const [guildId, array] of guildIdMap.entries()) {
+                const guildCommandURL = new URL(`${appid}/guilds/${guildId}/commands`, BASE_URL);
+                const response = await fetch(guildCommandURL, {
+                    method: 'PUT',
+                    body: JSON.stringify(array),
+                    headers,
+                });
+                const result = await response.json();
+                if (response.ok) {
+                    this.logger.info({ message: guildId + " published succesfully" })
+                    responsesMap.set(guildId, result);
+                } else {
+                    switch(response.status) {
+                        case 400 : {
+                            console.error(inspect(result, { depth: Infinity }))
+                            console.error("Modules with validation errors:" 
+                            + inspect(Object.keys(result.errors).map(idx => array[idx as any])))
+                            throw Error("400: Ensure your commands have proper fields and data and nothing left out");
+                        }
+                        case 404 : {
+                            console.error(inspect(result, { depth: Infinity }))
+                            throw Error("Forbidden 404. Is you application id and/or token correct?")
+                        }
+                        case 429: {
+                            console.error(inspect(result, { depth: Infinity }))
+                            throw Error('Chill out homie, too many requests')
+                        }
+                    }
+                }
+            }
+            await writeFile(
+                '.sern/command-data-remote.json',
+                JSON.stringify({ global: globalJsonBody,
+                               ...Object.fromEntries(responsesMap) }, null, 4),
+                'utf8')
+        }
+        this.sernEmitter.addListener('modulesLoaded', () => { 
+            listener(); 
+            this.sernEmitter.removeListener('modulesLoaded', listener);
+        })
+    }
+}
+
+export type ValidMemberPermissions = 
+    | typeof PermissionFlagsBits  //discord.js enum
+    | Array<typeof PermissionFlagsBits> 
+    | string //must be a stringified number
+    | bigint
+
+export interface PublishConfig {
+    guildIds?: string[];
+    defaultMemberPermissions?: ValidMemberPermissions;
+    integrationTypes?: Array<'Guild'|'User'>
```

Comment:
what does this mean?

Reply:
integrationTypes allows users to publish to a user only or dms